text/0000-more-api-documentation-conventions.md

```diff
@@ -0,0 +1,409 @@
+
+## Form
+[form]: #form
+
+These conventions relate to the formatting of the documentation, how they
+appear in source code.
+
+### Use line comments
+[use-line-comments]: #use-line-comments
+
+Avoid block comments. Use line comments instead:
+
+```rust
+// Wait for the main task to return, and set the process error code
+// appropriately.
+```
+
+Instead of:
+
+```rust
+/*
+ * Wait for the main task to return, and set the process error code
+ * appropriately.
+ */
+```
+
+Only use inner doc comments `//!` to write crate and module-level documentation,
+nothing else. When using `mod` blocks, prefer `///` outside of the block:
+
+```rust
+/// This module contains tests
+mod test {
+    // ...
+}
+```
+
+over
+
+```rust
+mod test {
+    //! This module contains tests
+
+    // ...
+}
+```
+
+### Using Markdown
+[using-markdown]: #using-markdown
+
+Within doc comments, use Markdown to format your documentation.
+
+Use top level headings # to indicate sections within your comment. Common headings:
+
+* Examples
+* Panics
+* Errors
+* Safety
+* Aborts
+* Undefined Behavior
+
+Even if you only include one example, use the plural form: ‘Examples’ rather
+than ‘Example’. Future tooling is easier this way.
+
+Use graves (`) to denote a code fragment within a sentence.
+
+Use triple graves (```) to write longer examples, like this:
+
+    This code does something cool.
+
+    ```rust
+    let x = foo();
+
+    x.bar();
+    ```
+
+When appropriate, make use of Rustdoc’s modifiers. Annotate triple grave blocks with
+the appropriate formatting directive.
+
+    ```rust
+    println!("Hello, world!");
+    ```
+
+    ```ruby
+    puts "Hello"
+    ```
+
+In API documentation, feel free to rely on the default being ‘rust’:
+
+    /// For example:
+    ///
+    /// ```
+    /// let x = 5;
+    /// ```
+
+In long-form documentation, always be explicit:
+
+    For example:
+
+    ```rust
+    let x = 5;
+    ```
+
+This will highlight syntax in places that do not default to ‘rust’, like GitHub.
+
+Rustdoc is able to test all Rust examples embedded inside of documentation, so
+it’s important to mark what is not Rust so your tests don’t fail.
+
+References and citation should be linked ‘reference style.’ Prefer
+
+```
+[Rust website]
+
+[Rust website]: http://www.rust-lang.org
+```
+
+to
+
+```
+[Rust website](http://www.rust-lang.org)
+```
+
+### Examples in API docs
+[examples-in-api-docs]: #examples-in-api-docs
+
+Everything should have examples. Here is an example of how to do examples:
+
+```
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// use op;
+///
+/// let s = "foo";
+/// let answer = op::compare(s, "bar");
+/// ```
+///
+/// Passing a closure to compare with, rather than a string:
+///
+/// ```
+/// use op;
+///
+/// let s = "foo";
+/// let answer = op::compare(s, |a| a.chars().is_whitespace().all());
+/// ```
+```
+
+For particularly simple APIs, still say “Examples” and “Basic usage:” for
+consistency’s sake.
```

Comment:
I do not like this practice for it is adds noise. Including "Basic usage" is sensible only when there is another example showing some other usage. Also, most of the examples (at least in the stdlib) are simple, so the consistency argument falls away.